test/common/router/router_test.cc

```diff
@@ -2763,23 +2765,359 @@ TEST_F(RouterTest, RetryRequestNotComplete) {
             callbacks.onPoolReady(encoder1, cm_.conn_pool_.host_, upstream_stream_info_);
             return nullptr;
           }));
-  EXPECT_CALL(callbacks_.stream_info_,
-              setResponseFlag(StreamInfo::ResponseFlag::UpstreamRemoteReset));
-  EXPECT_CALL(callbacks_.stream_info_, onUpstreamHostSelected(_))
-      .WillOnce(Invoke([&](const Upstream::HostDescriptionConstSharedPtr host) -> void {
-        EXPECT_EQ(host_address_, host->address());
+  expectResponseTimerCreate();
+
+  Http::TestRequestHeaderMapImpl headers{
```

Comment:
Do you think we should add this to the base test class, since so many tests use it?

Reply:
There are a lot of variations on the request headers to configure different things in these tests.

I thought about writing a base class or helper function for the several similar tests I'm adding in this PR, but it was becoming more confusing to follow what each test did, even though it reduced duplication. So I'm inclined to leave it as-is.